### PR TITLE
Fix remaining mypy/numpy errors

### PR DESCRIPTION
**Pull request by pavoljuhas**
- Fix type mismatch in _BufferedStateVector.create argument
- Fix type of the measurements argument to CliffordState.apply_measurement
- Fix type spec of the dtype argument to StateVectorSimulationState

`check/mypy --next` now passes without errors.

### CODE REVIEW

**Comment by pavoljuhas** [comment]
Partially implements #3767

File: cirq-core/cirq/testing/consistent_act_on.py [hunk 1]
```diff
@@ -46,12 +46,15 @@ def state_vector_has_stabilizer(state_vector: np.ndarray, stabilizer: DensePauli
...
+    complex_dtype: Type[np.complexfloating] = np.complex64
+    if np.issubdtype(state_vector.dtype, np.complexfloating):
+        complex_dtype = cast(Type[np.complexfloating], state_vector.dtype)
     args = state_vector_simulation_state.StateVectorSimulationState(
         available_buffer=np.empty_like(state_vector),
         qubits=qubits,
         prng=np.random.RandomState(),
         initial_state=state_vector.copy(),
-        dtype=state_vector.dtype,
+        dtype=complex_dtype,
```
Comment (dabacon) [inline cirq-core/cirq/testing/consistent_act_on.py:57]: in this case, since this is in the simulation direct call path, should we just do a cast here?

**Review by dabacon — CHANGES_REQUESTED** [review]
One question.

File: cirq-core/cirq/testing/consistent_act_on.py (hunk 1 above)
Comment (pavoljuhas) [reply cirq-core/cirq/testing/consistent_act_on.py:57]: Are we sure `state_vector.dtype` is complex?
If so we can make it simpler with cast.

**Review by dabacon — APPROVED** [review]
LGTM

**Pull request merged**